### DISCUSSION TITLE
Adding a proper dbus communication for plugins

### DISCUSSION THREAD

**Discussion by luxluth** [hyprwm/Hyprland#4279] [General]
### Why ?

Plugins can only be written in cpp. But any programming language that can interface with dbus can be used to develop plugins.

Most of us already have `dbus` installed somewhere on our system (for notification for example).

![hyperland-dbus](https://github.com/hyprwm/Hyprland/assets/74856540/d1fdc7b5-f651-443c-8dd8-eaf44addc85e)

**Comment by vaxerski** [comment]
zero point, we have socket and socket2 which do exactly the same. Plugins are for those that _need more_ and that can only be done in C++ because hl is written in C++.

**Comment by luxluth** [comment]
We'll it ever be possible to use other languages ?

**Reply by vaxerski** [reply]
no. Some rust fanboys even tried and failed miserably. Using other languages nullifies all the pros of having a plugin instead of a bash script.

You can write scripts / applications that utilize socket/socket2 (like hyprctl or socat for example do) in any language, though.

**Comment by luxluth** [comment]
Ok I see
